theano/scalar/basic.py

```diff
@@ -1928,7 +1928,14 @@ def c_code(self, node, name, inputs, outputs, sub):
 
         t = node.inputs[0].type.upcast(*[i.type for i in node.inputs[1:]])
         if t in imap(str, discrete_types):
-            check = 'if (%(y)s == 0) {PyErr_SetString(PyExc_ZeroDivisionError, "integer division by zero"); %(fail)s}' % locals()
+            check = dedent('''
+                if (%(y)s == 0) {
+                // do not set Python error message in gpuarray kernel for now
+                #ifndef LID_0
```

Comment:
I think there is a better macro to check for then this one. But I'm not sure if there is one for CUDA and a different one for OpenCL. @abergeron should know that.